generator/src/main/java/viper/generator/ConfigurationKeyProcessor.java

```diff
@@ -128,16 +138,38 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
 						note(e, "Generating property file based configuration resolver: " + configurationResolverClassName);
 						generateSourceFileFromTemplate(e, configurationResolverClassName, "ConfigurationResolver.vm", props);
 					}
+				} catch(IllegalArgumentException iae){
+					error(e, "Invalid paramters: "+ iae.getMessage());
 				} catch (IOException e1) {
-					processingEnv.getMessager().printMessage(Kind.ERROR, "Error creating files: " + Throwables.getStackTraceAsString(e1), e);
+					error(e, "Error creating files: " + Throwables.getStackTraceAsString(e1));
 				}
 			} else {
-				processingEnv.getMessager().printMessage(Kind.ERROR, "Code generation is supported only on enum types", e);
+				error(e, "Code generation is supported only on enum types");
 			}
 		}
 		return true;
 	}
 
+	static String calculateConfigurationBeanName(String className, String configurationBeanName) {
+		if (className == null) {
+			throw new IllegalArgumentException("Invalid class name");
+		}
+		if (configurationBeanName == null || !NAME_PATTERNS.matcher(configurationBeanName).matches()) {
+			throw new IllegalArgumentException("Invalid configurationBeanName pattern");
+		}
+		return configurationBeanName.replace("*", className);
+	}
+
+	static String calculateAnnotationName(String className, String annotationName) {
+		if (className == null) {
+			throw new IllegalArgumentException("Invalid class name");
+		}
+		if (annotationName == null || !NAME_PATTERNS.matcher(annotationName).matches()) {
+			throw new IllegalArgumentException("Invalid annotationName pattern");
+		}
+		return annotationName.replace("*", className);
```

Comment:
What happen if configurationBeanName does not contains any star symbol?

Reply:
Same as above